Packs/VersaDirector/Integrations/VersaDirector/VersaDirector.py

```diff
@@ -0,0 +1,3546 @@
+            t=response,
+            headerTransform=pascalToSpace,
+            headers=[
+                "app-name",
+                "app-risk",
+                "app-productivity",
+                "app-timeout",
+                "app-final-with-endpoint",
+            ],
+            is_auto_json_transform=True,
+            removeNull=True,
+        ),
+    )
+    return command_results
+
+
+def predefined_application_list_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    family = args.get("family")
+    risks = arg_to_number(args.get("risks"))
+    tags = argToList(args.get("tags"))
+    page = arg_to_number(args.get("page"))
+    page_size = arg_to_number(args.get("page_size"))
+    limit = arg_to_number(args.get("limit"))
+    offset = set_offset(page, page_size)
+
+    check_limit(limit)
+
+    response = client.predefined_application_list_request(family, risks, tags, offset, limit)
+
+    command_results = CommandResults(
+        outputs_prefix="VersaDirector.PredefinedApplication",
+        outputs_key_field="",
+        outputs=response,
+        raw_response=response,
+        readable_output=tableToMarkdown(
+            name="User predefined application objects.",
+            t=response,
+            headerTransform=pascalToSpace,
+            headers=[
+                "name",
+                "family",
+                "subfamily",
+                "description",
+                "risk",
+                "productivity",
+                "tag",
+            ],
+            is_auto_json_transform=True,
+            removeNull=True,
+        ),
+    )
+    return command_results
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+    try:
+        client._http_request(method="GET", url_suffix="vnms/alarm/notification", params={"limit": 25, "offset": 0})
+        message = "ok"
+    except DemistoException as e:
+        if "Forbidden" in str(e) or "Authorization" in str(e):
+            message = "Authorization Error: make sure authentication credentials are correctly set"
+        else:
+            raise e
+    return message
+
+
+# HEADING: """ MAIN FUNCTION """
+
+
+def main() -> None:
+    params: Dict[str, Any] = demisto.params()
+    args: Dict[str, Any] = demisto.args()
+    verify_certificate: bool = not params.get("insecure", False)
+    proxy = params.get("proxy", False)
+    use_token = params.get("use_token", False)
+    port = PORT_AUTH if use_token else PORT_CREDENTIALS
+    url = params.get("url", "") + f":{port}"
```

Comment:
You are right about checking the case where the url ends with "/", but using urljoin is not a good solution because the url concatenates with the port via ":" rather than "/".
Instead, I added a check to see if the URL ends with "/" before concatenating ":{port}".